src/NodaTime/DateInterval.cs

```diff
@@ -204,7 +204,7 @@ public void Deconstruct(out LocalDate start, out LocalDate end)
         /// </returns>
         /// <exception cref="ArgumentException"><paramref name="interval" /> uses a different
         /// calendar to this date interval.</exception>
-        public DateInterval? Intersection(DateInterval interval) =>
+        public DateInterval? Intersection([ValidatedNotNull] DateInterval interval) =>
```

Comment:
Odd, but I assume that this one _is_ still needed even though the first thing that happens is to pass `interval` to a methods with `[ValidatedNotNull]`?

(That is, we can't just stick this on the lower-level methods like `ValidateInterval()` and be done with it.)